src/modules/cpu/kernel/normalize.hpp

```diff
@@ -0,0 +1,340 @@
+            srcPtrTempRow += 8;
+            dstPtrTempRow += 8;
+        }
+        for(; vectorLoopCount < dims[1] ; vectorLoopCount += 8)
+             *dstPtrTempRow++ = (*srcPtrTempRow++ - mean) * invStdDev + shift;
+    }
+}
+
+RppStatus normalize_audio_host_tensor(Rpp32f* srcPtr,
+                                      RpptDescPtr srcDescPtr,
+                                      Rpp32f* dstPtr,
+                                      RpptDescPtr dstDescPtr,
+                                      Rpp32s *srcLengthTensor,
+                                      Rpp32s *channelsTensor,
+                                      Rpp32s axis_mask,
+                                      Rpp32f mean,
+                                      Rpp32f stdDev,
+                                      Rpp32f scale,
+                                      Rpp32f shift,
+                                      Rpp32f epsilon,
+                                      Rpp32s ddof,
+                                      Rpp32u numOfDims,
+                                      rpp::Handle& handle)
+{
+	Rpp32u numThreads = handle.GetNumThreads();
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+	for(int batchCount = 0; batchCount < srcDescPtr->n; batchCount++)
+	{
+        Rpp32f *srcPtrTemp = srcPtr + batchCount * srcDescPtr->strides.nStride;
+		Rpp32f *dstPtrTemp = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        // Set all values in dst buffer to 0.0
+        for(int cnt = 0; cnt < dstDescPtr->strides.nStride; cnt++)
+            dstPtrTemp[cnt] = 0.0f;
+
+        Rpp32u srcAudioDims[numOfDims], srcReductionDims[numOfDims], srcStride[numOfDims], paramStride[numOfDims];
+        srcAudioDims[0] = srcLengthTensor[batchCount];
+        srcAudioDims[1] = channelsTensor[batchCount];
+        if (axis_mask == 3) {
+            srcStride[0] = srcStride[1] = srcDescPtr->strides.cStride;
+            srcReductionDims[0] = 1;
+            srcReductionDims[1] = srcAudioDims[0] * srcAudioDims[1];
+            paramStride[0] = paramStride[1] = 0;
+        } else if (axis_mask == 1) {
+            srcStride[0] = (dstDescPtr->h > 1 and dstDescPtr->w > 1) ? srcDescPtr->strides.hStride : srcDescPtr->strides.wStride;
+            srcStride[1] = srcDescPtr->strides.cStride;
+            srcReductionDims[0] = srcAudioDims[1];
+            srcReductionDims[1] = srcAudioDims[0];
+            paramStride[0] = 1;
+            paramStride[1] = 0;
+        } else if (axis_mask == 2) {
+            srcStride[0] = srcDescPtr->strides.cStride;
+            srcStride[1] = (dstDescPtr->h > 1 and dstDescPtr->w > 1) ? srcDescPtr->strides.hStride : srcDescPtr->strides.wStride;
+            srcReductionDims[0] = srcAudioDims[0];
+            srcReductionDims[1] = srcAudioDims[1];
+            paramStride[0] = 0;
+            paramStride[1] = 1;
+        }
+
+        Rpp32f* meanTensor = (Rpp32f *)malloc(srcReductionDims[0] * sizeof(Rpp32f));
```

Comment:
Currently we are doing malloc and free for each sample in a loop which might be costly
Please try to allocate bigger buffer outside this for loop and free them outside the loop
See if this gives any improvement

Reply:
Initially I had tried with setting max buffer with max of both srcLength and channels but after setting this as per mask, this gave slight improvements too, so incorporated this also.